telegram/ext/_commandhandler.py

```diff
@@ -155,6 +159,23 @@ def collect_additional_context(
             if isinstance(check_result[1], dict):
                 context.update(check_result[1])
 
+    @property
+    def commands(self) -> Set[str]:
+        """
+        The set of commands this handler should listen for.
+
+        Returns:
+            Set[:obj:`str`]
+        """
+        return self._commands
+
+    @commands.setter
+    def commands(self, command: Union[str, List[str], Set[str]]) -> None:
+        if isinstance(command, str):
+            self._commands = {command.lower()}
+        else:
+            self._commands = {i.lower() for i in command}
```

Comment:
THis is the same logic that's used on `__init__`. Could you extract it to a new protected method (e.g. `_parse_commands()`) that you call both here and on `__init__`? Then you can also move the regexing from `__init__` to that function. THat will make sure that also the setter only accepts valid bot commands (please add a unit test for this)

Reply:
Sure, I first tried to do it like this
`self.commands = command`
But it wouldn't work when PrefixHandler tried to call it because it has a different setter than CommandHandler with the same name. But I'll make a function for it.

Reply:
I noticed we don't have CommandHandler command limitations for PrefixHandler, so illegal commands in prefixhandler should work properly shouldn't they?

Reply:
> I noticed we don't have CommandHandler command limitations for PrefixHandler, so illegal commands in prefixhandler should work properly shouldn't they?

yes, since `PrefixHandler` is not limited to the commands recognized by Telegram. Actually the fact that `PrefixHandler` is a subclass of `CommandHandler` is more or less an implementation detail …